modules/nlu/src/backend/election/natural-election.ts

```diff
@@ -45,20 +47,30 @@ function electIntent(input: sdk.IO.EventUnderstanding): sdk.IO.EventUnderstandin
   }
 }
 
-function extractElectedIntentSlot(input: sdk.IO.EventUnderstanding): sdk.IO.EventUnderstanding {
-  if (!input.predictions) {
+export function extractElectedIntentSlot(input: sdk.IO.EventUnderstanding): sdk.IO.EventUnderstanding {
+  if (isNil(input.predictions)) {
+    return input
+  }
+
+  const elected = input.intent
+  if (isNil(elected)) {
     return input
   }
 
-  const elected = input.intent!
+  const electedContext = input.predictions[elected.context]
+  if (isNil(electedContext)) {
+    return input
+  }
 
-  if (!elected) {
+  const electedContextIntents = electedContext.intents
+  if (isNil(electedContextIntents)) {
```

Comment:
If my understanding of the problem is right, this check should not be needed ?

Reply:
Done